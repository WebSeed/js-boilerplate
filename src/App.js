import React from 'react'

const App = () => (
  <div>
    <p>Whoop!</p>
  </div>
)

export default App
