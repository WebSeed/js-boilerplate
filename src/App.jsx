import React from 'react'

import image from './image.svg'

const waitFor = time =>
  new Promise(resolve => {
    setTimeout(() => {
      resolve(`Finished wait (${time}ms)`)
    }, time)
  })

const wait = async () => {
  const result = await waitFor(1000)
  alert(result)
}

const App = () => (
  <div className="app">
    <h1>It is working!</h1>
    <div className="image">
      <img src={image} alt="LBA" style={{ width: '240px' }} />
    </div>
  </div>
)

export default App
