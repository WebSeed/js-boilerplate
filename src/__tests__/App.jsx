import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import App from '../App'

const setup = (testProps = {}) => {
  const props = {
    ...testProps,
  }

  const wrapper = shallow(<App {...props} />)

  return {
    props,
    wrapper,
  }
}

describe('App', () => {
  it('renders matching snapshot', () => {
    const { wrapper } = setup()

    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
