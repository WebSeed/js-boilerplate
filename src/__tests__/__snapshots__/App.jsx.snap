// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders matching snapshot 1`] = `
<div
  className="app"
>
  <h1>
    It is working!
  </h1>
  <div
    className="image"
  >
    <img
      alt="LBA"
      src="file-mock"
      style={
        Object {
          "width": "240px",
        }
      }
    />
  </div>
</div>
`;
